opsdroid/connector/websocket/__init__.py

```diff
@@ -108,6 +153,20 @@ async def websocket_handler(self, request):
 
         return websocket
 
+    async def validate_request(self, request):
+        """Validate the request by looking at headers and the connector token.
+
+        If the token does not exist in the header, but exists in the configuration,
+        then we will simply return a Forbidden error.
+
+        """
+        client_token = request.headers.get("Authorization")
+        if self.authorization_token and (
+            client_token is None or client_token != self.authorization_token
+        ):
+            raise aiohttp.web.HTTPForbidden()
```

Comment:
In the case if the authorization failed I'd suggest to return HTTP 401 (Unauthorized) instead of 403 (Forbidden). The later means the authorization is successful but the user has no rights to access the resource.
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors

Reply:
That makes sense 😄  Thank you for the review updated the error with the Unauthorized one 👍